apps/question-the-docs/backend/ai/utils/text.py

```diff
@@ -15,22 +17,60 @@ class TextChunker:
     def __call__(self, src, files, cache):
         return pd.concat([self._slide_window(src, f, cache) for f in files])
 
+    def _get_code_block_lines(self, lines, non_ended_code_block):
+        lines = lines.to_numpy().tolist()
+        code_start = -1
+        code_end = -1
+        for i, line in enumerate(lines):
+            if any([lang for lang in CODE_BLOCK_PREFIX if f'```{lang}' in line]):
+                code_start = i
+            elif line.startswith('```'):
+                code_end = i
+
+        if code_end != -1:
+            non_ended_code_block = False
+
+        if code_end == -1 and code_start != -1:
+            non_ended_code_block = True
+
+        code_end = len(lines) if code_end == -1 else code_end
+        code_start = 0 if code_start == -1 else code_start
+
+        window_text = self.combine.join(lines)
+        excluded_text = ''
+        if not non_ended_code_block:
+            excluded_text = window_text
```

Comment:
Consider the example of `code_start=2` and `code_end=4`.  Then `not non_ended_code_block` is `True` and so we enter this `if` block. That means we include all the text. But what about the code between lines 2 and 4 - do we not want to remove this?

Reply:
the example is not possible 

non_ended_code_block means previous window had a non ended code block and so 

code_start=2 is not possible it will be code_start=-1 and code_end=4